File: ads/sorting algorithms/python/selectionSort.py
from helpers.swap import swap

def selectionSort(arr:list[float]):
    length = len(arr)

    for i in range(length-1):
        min_idx = i

        for j in range(i+1,length):
            if(arr[min_idx] > arr[j]):
                min_idx = j

        swap(arr,i,min_idx)

        
            

File: ads/sorting algorithms/c/selectionSort.c
void selectionSort(float arr[], int length)
{
    for (int i = 0; i < length; i++)
    {
        int idx_of_min = i;

        for (int j = i + 1; j < length; j++)
            if (arr[idx_of_min] > arr[j])
                idx_of_min = j;

        swap(&arr[idx_of_min], &arr[i]);
    }
}
